src/net/sf/mpxj/TaskContainer.java

```diff
@@ -62,6 +62,10 @@ public Task add()
     */
    @Override public void removed(Task task)
    {
+      //If the removal of attached data is suppressed do nothing
```

Comment:
I've not taken the changes to this file as I'm not sure what the use case is. We can deal with this one separately.

Reply:
The use case is the following:
When you create a project structure programmatically, then it might be that your source structures the Tasks by type. Summary tasks in one table Milestones in another, etc. and connected to those are Resources and Relations. The order of your Task creation does not conform to it's hierarchical structure. The default ordering in Project is by TaskID though, so that's why we added the method synchronizeTaskIDToHierarchy() earlier. Later I noticed (might be after a refactor), that this removed information attached to that task. hence the possibility to suppress this behaviour.